src/components/Table/DataTable/compenents/el-data-table/utils/select-strategy.js

```diff
@@ -159,18 +179,14 @@ class StrategyPersistSelection extends StrategyAbstract {
   updateElTableSelection() {
     const { data, id, selected } = this.elDataTable
 
-    // 历史勾选的行已经不在当前页了，所以要将当前页的行数据和selected合并
-    const mergeData = _.uniqWith([...data, ...selected], _.isEqual)
-
-    mergeData.forEach(r => {
-      const isSelected = !!selected.find(r2 => r[id] === r2[id])
+    if (!this.elTable) {
+      return Promise.resolve()
+    }
 
-      if (!this.elTable) {
-        return
-      }
+    const selectedIds = new Set(selected.map(r => r[id]))
+    const currentPageSelectedRows = data.filter(r => selectedIds.has(r[id]))
 
-      this.elTable.toggleRowSelection(r, isSelected)
-    })
+    return this.batchProcessSelection(currentPageSelectedRows)
   }
 }
 
```

Comment:
The provided code seems to be the implementation of several multi-select strategy classes (`StrategyNormal`, `StrategyPersistSelection`) within an entity. There appear to be no obvious errors or major discrepancies between the current version (as per my knowledge cut-off on September 1, 2021), but I can point out some small optimizations that might improve readability and maintenance efficiency:

1. Method names for `.on` events are not capitalized like they should be according to Java naming conventions.

2. The method body starts with a period which suggests it's intended to serve as a docstring for its function, rather than simply adding documentation directly underlined at the top.

3. Use destructuring in conditions could make them more readable. For example, instead of checking against the entire array inside a conditionals:

```javascript
const selectedIds = new Set(selected.map(r => r[id]));
```
This way, all elements can be found directly through their indices after selecting. 

Please consult the updated list from Google/IBM/OpenAI for specific coding standards changes relevant to these changes.